### DISCUSSION TITLE
Tire Pressure not showing up

### DISCUSSION THREAD

**Discussion by konstk1** [teslamate-org/teslamate#2841] [Q&A]
I'm on teslamate 1.27.1 and still not seeing tire pressure in grafana. The tire pressure time series in drive detail doesn't have any values. I've tried looking at several different trips.

**Comment by cwanja** [comment]
Are these new or historical drives?

**Reply by konstk1** [reply]
All new

**Reply by cwanja** [reply]
Do any of the logs show it’s being captured?

**Reply by konstk1** [reply]
what's the best way to check that?

**Reply by cwanja** [reply]
`docker-compose logs teslamate`

**Reply by konstk1** [reply]
There's a ton of stuff. What should I grep for? If I grep for tpms, it all comes back as 0 when it's attempting mqqt

```
teslamate  |     Args: [#Function<4.122580145/1 in TeslaMate.Mqtt.PubSub.VehicleSubscriber.handle_info/2>, [tpms_pressure_fl: 0.0]]
```

**Reply by cwanja** [reply]
That is the right value. In that example, the API is not reporting the TPMS values, which means TeslaMate cannot store and then graph them. You could experiment some more with polling the API directly, bypassing TeslaMate, and see if the API gives you the values. 

You would be looking for
```
tpms_pressure_fl,
       :tpms_pressure_fr,
       :tpms_pressure_rl,
       :tpms_pressure_rr
```

You should also have the car driving while polling.

**Reply by konstk1** [reply]
Is there a utility to do that?

**Reply by cwanja** [reply]
Not that I know of. You could start a session and poll the API, walk away, drive and come back. 

I just did a drive and it pulled the tire pressures for all four, so TeslaMate fundamentally is working to capture the readings, as long as they are available. On 1.27.1 as well.

**Reply by konstk1** [reply]
Doesn't look like the API is returning tire pressure.

```
teslamate  | 2022-08-16 08:09:46.187 [info] Start of drive initiated by: %TeslaApi.Vehicle.State.Drive{gps_as_of: 1660651785, heading: 158, latitude: 42.412132, longitude: -71.14207, native_latitude: 42.412132, native_location_supported: 1, native_longitude: -71.14207, native_type: "wgs", power: 5, shift_state: "D", speed: 0, timestamp: 1660651778554}
```

**Reply by cwanja** [reply]
That is just what TeslaMate is feeding the logs, not what is holistically being captured. Do you know how to poll the owners API yourself? https://tesla-api.timdorr.com from there, you could poll against the logs and see what the TPMS reports. Specifically https://tesla-api.timdorr.com/vehicle/state/vehiclestate

What is odd is that TeslaMate _is_ working, as others are not having TPMS data reporting issues (_raises hand_). So it would point to your configuration / streaming.

What car do you have as well?

**Reply by konstk1** [reply]
2022 Model Y
